evap/staff/views.py

```diff
@@ -1014,21 +1027,28 @@ def evaluation_person_management(request, semester_id, evaluation_id):
                 if not errors:
                     save_import_file(excel_file, request.user.id, import_type)
 
-        elif 'import' in operation:
-            file_content = get_import_file_content_or_raise(request.user.id, import_type)
-            success_messages, warnings, __ = PersonImporter.process_file_content(import_type, evaluation, test_run=False, file_content=file_content)
-            delete_import_file(request.user.id, import_type)
+        else:
+            if 'import' in operation:
+                if 'replace' in operation:
+                    deleted_person_count, deletion_message = helper_delete_users_from_evaluation(evaluation, operation)
```

Comment:
This is duplicated. You could move it out one scope, so that the logic is like:
```python3
if "test:
    pass # do test stuff
else:
    additional_messages = []
    if "replace":
        ... = helper_delete_users_from_evaluation(...)  # before the actual operation: clear everything
        # change additional messages
      
    if "copy":
        pass # do copy stuff
    elif "import":
        pass # do import stuff
       
    # merge messages and additional messages, unconditionally
```
Then, we don't have to check for "replace" three times.

Also, I'd suggest moving everything inside a [database transaction](https://docs.djangoproject.com/en/3.2/topics/db/transactions/), to make sure that either
- existing users get removed and the new users get added automatically, or,
- nothing is changed

It would be weird if, for whatever reason, you click on "replace" and end up with an empty data set because something went wrong.